syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -25,6 +25,10 @@ var (
 	// appear to be in a known format, or it breaks the rules of that format,
 	// or when there are I/O errors reading the file.
 	errUnrecognizedFormat = errors.New("unrecognized file format")
+
+	// errNotGoExe is returned when a given executable file is valid but does
+	// not contain Go build information.
+	errNotGoExe = errors.New("not a Go executable")
```

Comment:
I don't think this will work since this is a different error instance than the actual error being raised. Does the stdlib expose this error to use directly?